packages/api/core/src/api/make.ts

```diff
@@ -247,25 +248,13 @@ export const listrMake = (
             }
 
             for (const maker of makers) {
+              const uniqMaker = maker();
               subRunner.add({
-                title: `Making a ${chalk.magenta(maker.name)} distributable for ${chalk.cyan(`${platform}/${targetArch}`)}`,
+                title: `Making a ${chalk.magenta(uniqMaker.name)} distributable for ${chalk.cyan(`${platform}/${targetArch}`)}`,
                 task: async () => {
                   try {
-                    /**
-                     * WARNING: DO NOT ATTEMPT TO PARALLELIZE MAKERS
-                     *
-                     * Currently it is assumed we have 1 maker per make call but that is
-                     * not enforced.  It is technically possible to have 1 maker be called
-                     * multiple times.  The "prepareConfig" method however implicitly
-                     * requires a lock that is not enforced.  There are two options:
-                     *
-                     *   * Provide makers a getConfig() method
-                     *   * Remove support for config being provided as a method
-                     *   * Change the entire API of maker from a single constructor to
-                     *     providing a MakerFactory
-                     */
-                    await Promise.resolve(maker.prepareConfig(targetArch));
-                    const artifacts = await maker.make({
+                    await Promise.resolve(uniqMaker.prepareConfig(targetArch));
```

Comment:
```suggestion
                    await uniqMaker.prepareConfig(targetArch);
```

Reply:
I think we have to keep this `Promise.resolve` to ensure we aren't awaiting a non-promise

Reply:
We're justing calling the `prepareConfig` method here, not the actual maker config function/object, so we're guaranteed to be `await`ing a function

Reply:
`prepareConfig` is not guaranteed to return a promise though, we aren't awaiting the function, we're awaiting it's return value

Reply:
But `prepareConfig` doesn't return anything 🤷🏼‍♂️

https://github.com/electron/forge/blob/main/packages/maker/base/src/Maker.ts#L72-L78

Anyway, I'm marking this as resolved, the `Promise.resolve` wouldn't hurt a thing even if it happened to be unnecessary

Reply:
> But prepareConfig doesn't return anything 🤷🏼‍♂️

It does, `async function` implicitly defines the return value as a promise in that specific implementation, `prepareConfig` can be overridden by custom makers that might _not_ return a promise, hence the wrapping in `Promise.resolve`

Reply:
> `async function` implicitly defines the return value as a promise

Yeah, I meant to say that the _promise returned by `prepareConfig`_ resolves to `undefined` / `void`, and we're `await`ing the returned promise, _not_ the value it resolves to.

> `prepareConfig` can be overridden by custom makers that might _not_ return a promise

True, but the overridden method would still be a `function` (`async` or not), and to the best of my knowledge a plain `function` can be `await`ed without any consequences (plus you _could_ technically have something like `const arr = await [1,2,3]`, in this case the `await` is just a no-op of sorts).

(Sorry for all the nagging over this, but at this point I'm suspecting I might be fundamentally misunderstanding something and just want to know what it is 😅)

Reply:
I think @erikian has a point here, `await Promise.resolve(...)` is actually the implicit behavior of `await expression`:  

> [The `expression` is resolved in the same way as `Promise.resolve()`: it's always converted to a native Promise and then awaited.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await#description)